Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -506,15 +506,16 @@ private static Branch getBranchInstance(@NonNull Context context, boolean isLive
                 branchReferral_.linkCache_.clear();
                 branchReferral_.requestQueue_.clear();
             }
-        }
-        branchReferral_.context_ = context.getApplicationContext();
 
-        /* If {@link Application} is instantiated register for activity life cycle events. */
-        if (context instanceof BranchApp) {
-            isAutoSessionMode_ = true;
-            branchReferral_.setActivityLifeCycleObserver((Application) context);
+            branchReferral_.context_ = context.getApplicationContext();
+            /* If {@link Application} is instantiated register for activity life cycle events. */
+            if (context instanceof BranchApp) {
```

Comment:
won't this mean that `setActivityLifeCycleObserver` is only called if they've subclassed our Application? That's not recommended as a standard flow: https://dev.branch.io/getting-started/sdk-integration-guide/guide/android/#enable-auto-session-management